### DISCUSSION TITLE
"Meta" object through the onRecordAuthRequest

### DISCUSSION THREAD

**Discussion by suelitonx** [pocketbase/pocketbase#6002] [Q&A] [answered]
I tried to update the username that comes from the "meta" object through the onRecordAuthRequest hook, but I couldn't. Obviously, it is possible to do this on the client side (updating the record after registration), but I wanted to do it "under the hood".

Can anyone help me?

```
onRecordAuthRequest((e) => {
    if(e.authMethod == "oauth2" && e.meta != null) {
        if(e.meta.isNew && e.meta.name != null && e.record != null)
        {
            try {
                record.set("name", e.meta.name.toString());
                console.log("Successfully set name: ", e.meta.name);
            } catch (error) {
                console.log(error);
            }
        }
    }
    e.next()
}, "users");
```

**Comment by ganigeorgiev** [comment, marked as answer]
`record.set` should be `e.record.set`. But a better alternative would be to use the auto fields map functionality that you can adjust from the collection OAuth2 settings (_"OAut2 full name" is `e.meta.name`_):

![oauth2_settings](https://i.imgur.com/6Rg8h74.png)

**Reply by suelitonx** [reply]
Even correcting it to "e.record" didn't work. However, the best way you suggested worked perfectly, thank you very much!

**Reply by ganigeorgiev** [reply]
It should work with e.record but you have to persist it.
